src/Plugin/Field/FieldWidget/Select2EntityReferenceWidget.php

```diff
@@ -232,20 +234,28 @@ protected static function prepareFieldValues(array $values, array $element) {
    */
   protected function getAutocreateBundle() {
     $bundle = NULL;
-    if ($this->getSelectionHandlerSetting('auto_create') && $target_bundles = $this->getSelectionHandlerSetting('target_bundles')) {
-      // If there's only one target bundle, use it.
-      if (count($target_bundles) == 1) {
-        $bundle = reset($target_bundles);
+    if ($this->getSelectionHandlerSetting('auto_create')) {
+      if ($target_bundles = $this->getSelectionHandlerSetting('target_bundles')) {
+        // If there's only one target bundle, use it.
+        if (count($target_bundles) == 1) {
+          $bundle = reset($target_bundles);
+        }
+        // Otherwise use the target bundle stored in selection handler settings.
+        elseif (!$bundle = $this->getSelectionHandlerSetting('auto_create_bundle')) {
+          // If no bundle has been set as auto create target means that there is
+          // an inconsistency in entity reference field settings.
+          trigger_error(sprintf(
+            "The 'Create referenced entities if they don't already exist' option is enabled but a specific destination bundle is not set. You should re-visit and fix the settings of the '%s' (%s) field.",
+            $this->fieldDefinition->getLabel(),
+            $this->fieldDefinition->getName()
+          ), E_USER_WARNING);
+        }
       }
-      // Otherwise use the target bundle stored in selection handler settings.
-      elseif (!$bundle = $this->getSelectionHandlerSetting('auto_create_bundle')) {
-        // If no bundle has been set as auto create target means that there is
-        // an inconsistency in entity reference field settings.
-        trigger_error(sprintf(
-          "The 'Create referenced entities if they don't already exist' option is enabled but a specific destination bundle is not set. You should re-visit and fix the settings of the '%s' (%s) field.",
-          $this->fieldDefinition->getLabel(),
-          $this->fieldDefinition->getName()
-        ), E_USER_WARNING);
+      else {
+        $entity_definition = $this->entityTypeManager->getDefinition($this->getFieldSetting('target_type'));
+        if (!$entity_definition->getBundleEntityType()) {
+          $bundle = $this->getFieldSetting('target_type');
```

Comment:
Instead of hard coding target_type, we might check in the actual entity what her bundle is.

Reply:
We don't have a real entity at that point, only an entity definition. And core uses the entityTypeId in all places if no bundle exists. e.g. ContentEntityBase::construct, EntityBase::bundle